### PR TITLE
feat(start_planner): define ignore section separately

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

launcher [PR](https://github.com/autowarefoundation/autoware_launch/pull/832) should be merged first

In this [PR](https://github.com/autowarefoundation/autoware.universe/pull/5982), apply same offset colission check section for shift/geometric pull out path.
But path length is different and 
- In some cases, ignore_section is necessary for shift pull out path but if same length of ignore section is applied to geometric pull out path, whole or almost all section of ignored and no collision check is executed.

image

So in this PR, define `shift_collision_check_distance_from_end` for shift pull out path `geometric_collision_check_distance_from_end` for geometric pull out path.

## Tests performed
With following default parameter, I confirmed collision check end pose is offset to shift pull out path but not to geometric pull out path.

Screenshot from 2024-01-30 09-31-40
geometric_collision_check_distance_from_end = 0.0

Screenshot from 2024-01-30 09-32-37
shift_collision_check_distance_from_end = -10.0

- [ ] [PASS TIERIV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/94868a09-4bb1-5551-ad00-a3fcaef34a5d?project_id=prd_jt)

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp [hunk 1]
```diff
@@ -672,8 +672,17 @@ bool StartPlannerModule::findPullOutPath(
...
+  const PullOutPath & path, const behavior_path_planner::PlannerType & planner_type)
 {
+  std::map<PlannerType, double> collision_check_distances = {
+    {behavior_path_planner::PlannerType::SHIFT,
+     parameters_->shift_collision_check_distance_from_end},
+    {behavior_path_planner::PlannerType::GEOMETRIC,
+     parameters_->geometric_collision_check_distance_from_end}};
+
+  double collision_check_distance_from_end = collision_check_distances[planner_type];
+
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/src/start_planner_module.cpp:685]:
```suggestion
  const std::map<PlannerType, double> collision_check_distances = {
    {behavior_path_planner::PlannerType::SHIFT,
     parameters_->shift_collision_check_distance_from_end},
    {behavior_path_planner::PlannerType::GEOMETRIC,
     parameters_->geometric_collision_check_distance_from_end}};

  const double collision_check_distance_from_end = collision_check_distances[planner_type];

```

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp [hunk 2]
```diff
@@ -1403,9 +1412,14 @@ void StartPlannerModule::setDebugData()
   // visualize collision_check_end_pose and footprint
   {
     const auto local_footprint = vehicle_info_.createFootprint();
+    std::map<PlannerType, double> collision_check_distances = {
+      {PlannerType::SHIFT, parameters_->shift_collision_check_distance_from_end},
+      {PlannerType::GEOMETRIC, parameters_->geometric_collision_check_distance_from_end}};
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/src/start_planner_module.cpp:1417]: (nits: this is duplicated process, maybe it is also good have as member)

Comment (kyoichi-sugahara) [reply planning/behavior_path_start_planner_module/src/start_planner_module.cpp:1417]: ah
define map as member variable you mean?
I agree

Comment (kosuke55) [reply planning/behavior_path_start_planner_module/src/start_planner_module.cpp:1417]: yes because this map is like a parameter

**Review by kosuke55 — APPROVED** [review]

**Pull request merged**